synapse/handlers/room_member.py

```diff
@@ -347,7 +347,17 @@ async def update_membership_locked(
             # later on.
             content = dict(content)
 
-        if not self.allow_per_room_profiles or requester.shadow_banned:
+        is_requester_server_notices_user = False
+        if (
+            self._server_notices_mxid is not None
+            and requester.user.to_string() == self._server_notices_mxid
+        ):
+            # allow the server notices mxid to set room-level profile
+            is_requester_server_notices_user = True
```

Comment:
No need for a conditional

```suggestion
        # allow the server notices mxid to set room-level profile
        is_requester_server_notices_user = (
            self._server_notices_mxid is not None
            and requester.user.to_string() == self._server_notices_mxid
        )
```